pkg/irreparable/irreparable.go

```diff
@@ -0,0 +1,174 @@
+// Copyright (C) 2018 Storj Labs, Inc.
+// See LICENSE for copying information.
+
+package irreparable
+
+import (
+	"context"
+	"time"
+
+	"go.uber.org/zap"
+	"google.golang.org/grpc/codes"
+	"google.golang.org/grpc/status"
+	monkit "gopkg.in/spacemonkeygo/monkit.v2"
+
+	"storj.io/storj/internal/migrate"
+	"storj.io/storj/pkg/auth"
+	dbx "storj.io/storj/pkg/irreparable/dbx"
+	pdbauth "storj.io/storj/pkg/pointerdb/auth"
+)
+
+var (
+	mon = monkit.Package()
+)
+
+// Server implements the irreparable RPC service
+type Server struct {
+	DB     *dbx.DB
+	logger *zap.Logger
+}
+
+// RemoteSegmentInfo is info about a single entry stored in the irreparable db
+type RemoteSegmentInfo struct {
+	Key                []byte `protobuf:"bytes,1,opt,name=key,proto3" json:"key,omitempty"`
+	Val                []byte `protobuf:"bytes,2,opt,name=val,proto3" json:"val,omitempty"`
+	LostPiecesCount    int64  `protobuf:"varint,3,opt,name=lost_pieces_count,json=lostPiecesCount,proto3" json:"lost_pieces_count,omitempty"`
+	RepairUnixSec      int64  `protobuf:"varint,4,opt,name=repair_unix_sec,json=repairUnixSec,proto3" json:"repair_unix_sec,omitempty"`
+	RepairAttemptCount int64  `protobuf:"varint,6,opt,name=repair_attempt_count,json=repairAttemptCount,proto3" json:"repair_attempt_count,omitempty"`
+}
+
+// NewServer creates instance of Server
+func NewServer(driver, source string, logger *zap.Logger) (*Server, error) {
+	db, err := dbx.Open(driver, source)
+	if err != nil {
+		return nil, err
+	}
+
+	err = migrate.Create("irreparable", db)
+	if err != nil {
+		return nil, err
+	}
+
+	return &Server{
+		DB:     db,
+		logger: logger,
+	}, nil
+}
+
+func (s *Server) validateAuth(ctx context.Context) error {
```

Comment:
This shouldn't have authentication. Databases are protected by passwords and firewalls.